core/core.js

```diff
@@ -189,63 +189,57 @@ SharedQueue Binary Layout
     return errorMap[errorName] ?? [undefined, []];
   }
 
-  // Returns Uint8Array
-  function encodeJson(args) {
-    const s = JSON.stringify(args);
-    return core.encode(s);
-  }
-
-  function decodeJson(ui8) {
-    const s = core.decode(ui8);
-    return JSON.parse(s);
-  }
-
   let nextPromiseId = 1;
-  const promiseTable = {};
+  const promiseTable = new Map();
```

Comment:
Warning, map was previously used to store promises, but was replaced with an object due to GC performance problems https://github.com/denoland/deno/pull/4309

Reply:
@bartlomieju I'll follow-up on this in a separate PR and revert it if `Object` ultimately beats `Map`. It will be easier to investigate this once we have baseline-overhead benches for async ops (follow-up PR `8.`)